Marlin/gcode.h

```diff
@@ -148,6 +169,9 @@ class GCodeParser {
   // Code value pointer was set
   FORCE_INLINE static bool has_value() { return value_ptr != NULL; }
 
+  // Seen and has value
+  FORCE_INLINE static bool seenval(const char c) { return seen(c) && has_value(); }
```

Comment:
Also, I anticipate replacing at least a 12-18 `seen(X) && has_value()` conditionals scattered throughout the codebase.  Didn't include them here because they're unrelated.